File: src-py/repositories/worker_repository.py
from utils.prisma import prisma


async def getWorkers() -> list:
    return await prisma.worker.find_many()


async def getWorkerById(id: int):
    return await prisma.worker.find_unique(where={"id": id})


async def getWorkerByRegistration(registration: int):
    return await prisma.worker.find_unique(where={"registration": registration})

File: server/repositories/workerRepository.ts
import prisma from "../utils/prisma"

export async function getWorkers() {
    try {
        const workers = await prisma.worker.findMany()
        return workers
    } catch (error) {
        throw error
    }
}

export async function getWorkerById(id: number) {
    try {
        const worker = await prisma.worker.findUnique({
            where: { id }
        })
        return worker
    } catch (error) {
        throw error
    }
}

export async function getWorkerByRegistration(registration: number) {
    try {
        const worker = await prisma.worker.findUnique({
            where: { registration }
        })
        return worker
    } catch (error) {
        throw error
    }
}
